packages/contracts-bedrock/test/L1/L1StandardBridge.t.sol

```diff
@@ -857,4 +1094,294 @@ contract L1StandardBridge_Unclassified_Test is L1StandardBridge_TestInit {
         l1StandardBridge.finalizeBridgeETH{ value: 100 }(alice, messenger, 100, hex"");
     }
+
+    /// @notice Prevents deposit accounting invariant violations across multiple operations
+    ///         that could allow attackers to manipulate total deposited balances.
+    function test_integration_depositWithdrawalInvariant_succeeds() external {
+        uint256 depositAmount = 1000;
+        uint256 withdrawAmount = 600;
+
+        // Setup ERC20 tokens
+        deal(address(L1Token), alice, depositAmount, true);
+        vm.prank(alice);
+        L1Token.approve(address(l1StandardBridge), depositAmount);
+
+        // Initial deposit
+        vm.prank(alice, alice);
+        l1StandardBridge.depositERC20(address(L1Token), address(L2Token), depositAmount, 50000, hex"");
+
+        uint256 depositsAfterDeposit = l1StandardBridge.deposits(address(L1Token), address(L2Token));
+        assertEq(depositsAfterDeposit, depositAmount);
+
+        // Simulate withdrawal
+        deal(address(L1Token), address(l1StandardBridge), depositAmount, true);
+
+        vm.mockCall(
+            address(l1StandardBridge.messenger()),
+            abi.encodeCall(ICrossDomainMessenger.xDomainMessageSender, ()),
+            abi.encode(address(l1StandardBridge.OTHER_BRIDGE()))
+        );
+
+        vm.prank(address(l1StandardBridge.messenger()));
+        l1StandardBridge.finalizeERC20Withdrawal(
+            address(L1Token), address(L2Token), alice, alice, withdrawAmount, hex""
+        );
+
+        uint256 depositsAfterWithdrawal = l1StandardBridge.deposits(address(L1Token), address(L2Token));
+        assertEq(depositsAfterWithdrawal, depositAmount - withdrawAmount);
+    }
+
+    /// @notice Prevents cross-domain message authentication bypass attacks by testing
+    ///         various malicious messenger scenarios that could spoof bridge calls.
+    /// @param _fakeSender Random address to test cross-domain authentication.
+    function testFuzz_finalizeBridgeETH_crossDomainAuthBypass_reverts(address _fakeSender) external {
+        vm.assume(_fakeSender != address(l1StandardBridge.OTHER_BRIDGE()));
+
+        vm.mockCall(
+            address(l1StandardBridge.messenger()),
+            abi.encodeCall(ICrossDomainMessenger.xDomainMessageSender, ()),
+            abi.encode(_fakeSender)
+        );
+
+        vm.deal(address(l1StandardBridge.messenger()), 100);
+
+        vm.prank(address(l1StandardBridge.messenger()));
+        vm.expectRevert("StandardBridge: function can only be called from the other bridge");
+        l1StandardBridge.finalizeBridgeETH{ value: 100 }(alice, alice, 100, hex"");
+    }
+
+    /// @notice Prevents EOA bypass attacks using various address types that could
+    ///         circumvent the onlyEOA modifier protection mechanism.
+    /// @param _caller Random address to test EOA restriction comprehensively.
+    function testFuzz_depositETH_eoaBypassProtection_reverts(address _caller) external {
```

Comment:
Same as above, seems like a bug, this clearly tests depositETH but got put into uncategorized